server/src/service.rs

```diff
@@ -99,9 +99,42 @@ impl Service {
     #[zbus(out_args("locked", "prompt"))]
     pub async fn lock(
         &mut self,
-        _objects: Vec<OwnedObjectPath>,
+        objects: Vec<OwnedObjectPath>,
+        #[zbus(connection)] connection: &zbus::Connection,
+        #[zbus(signal_emitter)] signal_emitter: SignalEmitter<'_>,
     ) -> Result<(Vec<OwnedObjectPath>, ObjectPath), ServiceError> {
-        todo!()
+        let mut locked: Vec<OwnedObjectPath> = Vec::new();
+        let prompt = ObjectPath::default();
```

Comment:
Yeah I agree. But, a prompt isn't required here, like at all. So, returning an empty objectpath: '/' is enough. And we don't have to worry about cleaning up later :)

Reply:
Based on three things:
* There is nothing on the spec about keeping track of the objects that are locked with a prompt and a way to return them. This can be consider as a hint there is nothing related to actually prompting a system promt here.
* gnome-keyring also returns an empty object path as the prompt: https://gitlab.gnome.org/GNOME/gnome-keyring/-/blob/master/daemon/dbus/gkd-secret-service.c#L609
* I have tested this call many time. And never found any use case involving a prompt.

Reply:
gnome-keyring-daemon is the least respectful for the secrets spec, i wouldn't follow it "blindly"

Reply:
> There is nothing on the spec about keeping track of the objects that are locked with a prompt and a way to return them. This can be consider as a hint there is nothing related to actually prompting a system prompt here.

There is, the prompt has a details return variable that is a zvariant::Value, it can contain anything, even a list of object paths ;)

Reply:
yes, there is no way to pass a list of items to unlock into prompt & then get the result of what got locked and what wasn't. But there should definitely be a prompt used here otherwise the spec is wrong which I don't think it is...